src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs

```diff
@@ -94,6 +94,10 @@ internal class CSharpRecommendationService : AbstractRecommendationService
             {
                 return GetSymbolsForNamespaceDeclarationNameContext(context, cancellationToken);
             }
+            else if (context.IsFirstDotOfDotDotToken)
+            {
+                return GetSymbolsForExpressionOrStatementContext(context, filterOutOfScopeLocals, cancellationToken);
```

Comment:
i feel like this is somewhat of an add place for this call.  i would have expected this to be nearer to the code that is getting symbols when teh user hits `dot`.  i.e. it would check the normal `dot` case *and* this specialized case.

Reply:
As I remember, the normal dot goes
to GetSymbolsOffOfContainer at 
[https://github.com/dotnet/roslyn/blob/fab7134296816fc80019c60b0f5bef7400cf23ea/src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs#L52](https://github.com/dotnet/roslyn/blob/fab7134296816fc80019c60b0f5bef7400cf23ea/src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs#L52)
due to IsRightOfNameSeparator == true.

So, I do not think we can move this dotdot to the normal dot. There is another syntax tree in case of a normal dot.

I have moved to some lines above and joined other conditions. Please let me know if you see a better opportunity.

---
In reply to: [254006890](https://github.com/dotnet/roslyn/pull/33148#discussion_r254006890) [](ancestors = 254006890)

Reply:
Sorry, i wasn't clear.  What i'm saying is:

I find it surprising that you made your edit *here*.  I would have expected this new code to be closer (proximity wise) to where the normal 'dot' logic goes.

Looking at the code you linked to, here's what i think might be a better fix:

1. get rid of "isafterdotofdotdot"
2. inside SyntaxContext, set IsRightOfNameSeparator to true if you're after the dot of a dotdot.

Conceptually, that's where you are if your cursor is here:  `foo.$$.Bar()`.